src/ape/managers/compilers.py

```diff
@@ -78,15 +85,18 @@ def compile(self, contract_filepaths: List[Path]) -> Dict[str, ContractType]:
             Dict[str, ``ContractType``]: A mapping of contract names to their type.
         """
 
+        # NOTE: Must cache 'base_path' before checking registered compilers.
```

Comment:
Could do `lrucache` using project_folder?

Reply:
`project_folder` is not an argument. We just need to grab the value (`self.config.contracts_folder`) before calling `registered_compilers`. Because at that point, if we are compiling a dependency, it will reload the config file and we will lose the path set by the `dependencies:` stanza in our root project config. This is only a problem when the dependency has a different named contracts folder than the default.